Topics/Development_Process/Trunk_Development.md

```diff
@@ -0,0 +1,5 @@
+### Trunk-Based Development
+
+Knowing how to manage the creation of large new features in your repository is an important skill, especially when working in large organizations or on projects with complete CI systems. It can be tempting to have large feature branches with hundreds of new lines of code and it can seem daunting to implement a large feature in small stages. Trunk-based development is a system to handle large changes like this. It focuses on creating small PR and using tools such as feature flags. The following article from Atlassian does a great job discussing all the benefits of trunk-based development (i.e. impact on automated testing, having fewer active branches, and more): **[Atlassian Article](https://www.atlassian.com/continuous-delivery/continuous-integration/trunk-based-development)**
+
+Instead of using feature flags, one can also use the "branch by abstraction" method found in [this article](https://trunkbaseddevelopment.com/branch-by-abstraction/). This is similar to referencing an interface instead of a class and much like dependency inversion, can both increase or reduce the complexity of your code.
```

Comment:
Maybe you can do a one-sentence elaboration on which situations to use it, since it can be a double edged sword in that in certain cases it may actually increase the complexity of your code.

Reply:
Will do! I am going to modify this idea a little by just converting the feature-flag method to abstraction and then saying the developer can choose which is better (because they can). Since this topic is not the focus, it will be relatively low detail.